src/app/legacy/components/ScrollablePromo/Promo/index.jsx

```diff
@@ -81,47 +102,96 @@ const TimeStamp = styled(PromoTimestamp)`
   color: ${({ theme }) => theme.isDarkUi && theme.palette.GREY_6};
 `;
 
-const Promo = ({ block, onClick }) => {
+const Promo = ({ block, experimentVariant, onClick }) => {
   const { script, service, serviceDatetimeLocale } = useContext(ServiceContext);
-  const textBlock = filterForBlockType(
-    pathOr({}, ['model', 'blocks'], block),
-    'text',
-  );
-  const aresLinkBlock = filterForBlockType(
-    pathOr({}, ['model', 'blocks'], block),
-    'aresLink',
-  );
-  const href = pathOr(
-    '',
-    ['model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'locator'],
-    textBlock,
-  );
-  const title = pathOr(
-    '',
-    ['model', 'blocks', '0', 'model', 'blocks', '0', 'model', 'text'],
-    textBlock,
-  );
-  const timestamp = path(
-    ['model', 'blocks', '0', 'model', 'timestamp'],
-    aresLinkBlock,
-  );
+  let title;
+  let href;
+  let textBlock;
+  let aresLinkBlock;
+  let timestamp;
+  let isLive;
+  switch (experimentVariant) {
+    case 'A': {
+      const overtypedHeadline = block?.headlines?.overtyped ?? '';
+      const mainHeadline = block?.headlines?.headline ?? '';
+      const headlineBlockText =
+        block?.headlines?.promoHeadline?.blocks?.[0]?.model?.blocks?.[0]?.model
+          ?.text ?? '';
+      const name = block?.name ?? '';
+
+      title =
+        overtypedHeadline ||
+        mainHeadline ||
+        headlineBlockText ||
+        name ||
+        block.headline ||
+        '';
+
+      const canonicalUrl = block?.locators?.canonicalUrl ?? '';
+      const assetUri = block?.locators?.assetUri ?? '';
+      const uri = block?.uri ?? '';
+
+      href =
+        canonicalUrl ||
+        assetUri ||
+        uri ||
+        (block.destinationUrl
+          ? `https://www.bbc.com${block.destinationUrl}`
+          : '');
+      isLive = block.isLive;
+      break;
+    }
+    case 'B':
+      title = block.title;
+      href = block.href;
+      break;
+    default:
+      textBlock = filterForBlockType(
+        pathOr({}, ['model', 'blocks'], block),
```

Comment:
Since we're modifying a legacy component here should we be removing Ramda functions like `path0r` and replacing with suitable alternatives? Does this apply to experiments?